### PR TITLE
APIS-4758 - Add PPNS Topic field type

### PR DESCRIPTION
**Pull request by AndySpaven**

### CODE REVIEW

File: app/uk/gov/hmrc/apisubscriptionfields/connector/JsonFormatters.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package uk.gov.hmrc.apisubscriptionfields.connector
+
+import play.api.libs.json._
+import uk.gov.hmrc.apisubscriptionfields.model.SpecialJsonFormatters
+
+trait JsonFormatters extends SpecialJsonFormatters {
+  implicit val createTopicRequestJF = Json.format[CreateTopicRequest]
+  implicit val createTopicResponsetJF = Json.format[CreateTopicResponse]
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apisubscriptionfields/connector/JsonFormatters.scala:24]: Extra t in the name

File: app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsRepository.scala [hunk 2]
```diff
@@ -93,43 +94,43 @@ class SubscriptionFieldsMongoRepository @Inject()(mongoDbProvider: MongoDbProvid
...
   override def delete(clientId: ClientId): Future[Boolean] = {
-    deleteMany(clientIdSelector(clientId.value))
+    deleteMany(clientIdSelector(clientId))
   }
 
-  private def clientIdSelector(clientId: String) = Json.obj("clientId" -> clientId)
+  private def clientIdSelector(clientId: ClientId) = Json.obj("clientId" -> clientId.value)
 
-  private def fieldsIdSelector(fieldsId: UUID) = Json.obj("fieldsId" -> fieldsId)
+  private def fieldsIdSelector(fieldsId: SubscriptionFieldsId) = Json.obj("fieldsId" -> fieldsId.value.toString)
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsRepository.scala:122]: should not have toString

File: conf/application.conf [hunk 3]
```diff
@@ -142,50 +144,28 @@ microservice {
...
 
-      delay-response = 2 seconds
+wiremock-port = 11111
+wiremock-port = ${?WIREMOCK_PORT}
 
-      protocol = http
-    }
+Dev {
+    mongodb {
+    uri = "mongodb://localhost:27017/api-subscription-fields-test?rm.monitorRefreshMS=4000&rm.failover=default"
```
Comment (AndySpaven) [inline conf/application.conf:169]: remove the excess confi

File: app/uk/gov/hmrc/apisubscriptionfields/connector/JsonFormatters.scala (hunk 1 above)
Comment (peteslater-ee) [reply app/uk/gov/hmrc/apisubscriptionfields/connector/JsonFormatters.scala]: I think this has gone now

File: conf/application.conf (hunk 3 above)
Comment (peteslater-ee) [reply conf/application.conf:175]: I think this is the same is it used to be so is OK

File: app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsRepository.scala (hunk 2 above)
Comment (peteslater-ee) [reply app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsRepository.scala]: This has gone now

**Pull request merged**